1️⃣ Estrutura do Dataset

In [2]:
import numpy as np
import pandas as pd

In [3]:
path_csv = r'/home/jvdata/Documentos/Case-SEGMEDIC/src/data/prata/Dados_limpos.csv'

- Carregar o data set

In [4]:
df_limpo = pd.read_csv(path_csv, sep=";")

- Exibir as 5 Primeiras Linhas

In [5]:
df_limpo.head(6)

,Data_Atendimento,ID_paciente,Nome_procedimento,Categoria_procedimento,valor_pago,retorno,Unidade,Metodo_pagamento,ID_profissional
0,2024-01-30,104593,Rx Tornozelo Esquerdo,Raio X,49.0,Não,Centro Medico Matriz,PIX,18252
1,2024-01-30,104593,Rx Joelho,Raio X,49.0,Não,Centro Medico Matriz,PIX,18252
2,2024-02-06,166899,Consulta Clinica Geral,Clinica Geral,19.0,Não,Centro Medico Matriz,PIX,18052
3,2024-02-22,188620,Consulta Clinica Geral,Clinica Geral,19.0,Não,Centro Medico Matriz,PIX,18190
4,2024-02-26,45111,Consulta Clinica Geral,Clinica Geral,19.0,Não,Centro Medico Matriz,PIX,18188
5,2024-03-27,18877,Consulta Clinica Geral,Clinica Geral,19.0,Não,Centro Medico Matriz,PIX,17955


- Quantas linhas e colunas existem no arquivo?

In [6]:
df_limpo.shape

(5050, 9)

Linhas = 5050
Colunas = 9

2️⃣ Valor Atendido por Mês

- Qual o valor total atendido por mês?

In [7]:
#houve um erro na mudança de data type da coluna. corrigir repetindo o processo de mudança de data type.

df_limpo['Data_Atendimento'] = pd.to_datetime(df_limpo['Data_Atendimento'])

In [8]:
df_limpo.dtypes

Data_Atendimento          datetime64[us]
ID_paciente                        int64
Nome_procedimento                    str
Categoria_procedimento               str
valor_pago                       float64
retorno                              str
Unidade                              str
Metodo_pagamento                     str
ID_profissional                    int64
dtype: object

In [9]:
valor_mes_agrupado = df_limpo.groupby(pd.Grouper(key='Data_Atendimento', freq='ME'))['valor_pago'].agg(['sum', 'mean'])

ME = Mounth End

In [10]:
valor_mes_agrupado = valor_mes_agrupado.rename(columns={
    'sum': 'Soma_Total', 
    'mean': 'Media_Mensal'
})
valor_mes_agrupado.head()

,Soma_Total,Media_Mensal
Data_Atendimento,,
2024-01-31,510263.75,1457.896429
2024-02-29,7544.13,25.924845
2024-03-31,12312.81,27.983659
2024-04-30,7627.93,25.683266
2024-05-31,9444.45,27.139224


Soma total:
- Janeiro = 510.263,75
- Fevereiro = 7.544,13
- Março = 12.312,81	
- Abril = 7.627,93	
- Maio = 9.444,45	

3️⃣ Quantidade de Atendimentos por Mês

- Quantos atendimentos ocorreram em cada mês?

In [13]:
atendimento_mes = df_limpo.groupby(pd.Grouper(key='Data_Atendimento', freq='ME'))['ID_paciente'].count()
atendimento_mes.head()

Data_Atendimento
2024-01-31    350
2024-02-29    291
2024-03-31    440
2024-04-30    297
2024-05-31    348
Freq: ME, Name: ID_paciente, dtype: int64

4️⃣ Pacientes com Maior Frequência